implementation/leprechaun/tsconfig.json

```diff
@@ -0,0 +1,21 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "declaration": true,
+    "removeComments": true,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "allowSyntheticDefaultImports": true,
+    "target": "ES2021",
+    "sourceMap": true,
+    "outDir": "./dist",
+    "baseUrl": "./",
+    "incremental": true,
+    "skipLibCheck": true,
+    "strictNullChecks": false,
```

Comment:
```suggestion
    "strictNullChecks": true,
```

We already have Java in the repository, let's not have its clone in JavaScript.

`const test: string = null;` passes with this set to `false` and doesn't with it set to `true`. To be honest, you should probably enable `strict` https://www.typescriptlang.org/tsconfig/#strict.

Reply:
Oh wow, no need to be offensive xD. Let's not start a classic war about languages. Especially since neither Java nor Js is in the best position to argue about type safety

Reply:
![Screenshot_20241229-151553.png](https://github.com/user-attachments/assets/c396b9ef-e936-437e-a670-0955a945869c)

![Screenshot_20241229-151620.png](https://github.com/user-attachments/assets/a13ecaa0-9f16-4594-9587-021b60def1b7)

![Screenshot_20241229-151706.png](https://github.com/user-attachments/assets/537ee964-dfb4-4114-8dfd-58ffda58fb46)

![Screenshot_20241229-151742.png](https://github.com/user-attachments/assets/c78a12ea-144d-4de4-aa0f-3055fb9949cf)

![Screenshot_20241229-151820.png](https://github.com/user-attachments/assets/e3a30e62-f104-41e4-868b-151c810f4149)

Reply:
I did
```jsonc
// ...
    "strict": true,
    "strictNullChecks": true,
    "noImplicitAny": true,
    "strictBindCallApply": true,
// ...
```
let's see how long till depression kicks in.

Reply:
nvm, i've started hearing voices. I'll make only `strict` true.

Reply:
> I'll make only strict true.

Strict enables many other settings and I'd say that it is probably enough for your needs.